### PR TITLE
Adds support for list values in Bite.ite

### PR DESCRIPTION
**Pull request by leonardt**
This allows you to assign lists inside combinational code
It simply calls ite on the zipped elements of the list inputs and packs
the result in a list.

### CODE REVIEW

**Comment by leonardt** [comment]
Also adds support for hwtypes.Bit/bool to mux/ite

File: magma/array.py [hunk 1]
```diff
@@ -214,12 +216,15 @@ def __init__(self, *args, **kwargs):
...
                     self.ts = args[0].ts[:]
-                elif isinstance(args[0], int):
+                elif isinstance(args[0], (BitVector, int)):
                     if not issubclass(self.T, Bit):
                         raise TypeError(f"Can only instantiate Array[N, Bit] "
-                                        f"with int, not Array[N, {self.T}]")
+                                        f"with int/bv, not Array[N, {self.T}]")
+                    n = args[0]
+                    bits = (int2seq(n, self.N) if isinstance(n, int)
+                            else n.bits())
```
Comment (rsetaluri) [inline magma/array.py:225]: nit:
```suggestion
                    bits = (int2seq(n, self.N)
                            if isinstance(n, int)
                            else n.bits())
```

File: magma/bit.py [hunk 2]
```diff
@@ -135,14 +135,21 @@ def __xor__(self, other):
...
+                raise TypeError("Bit.ite of two lists expects the same length")
+            return [self.ite(x, y) for x, y in zip(t_branch, f_branch)]
         t_type = get_type(t_branch)
         f_type = get_type(f_branch)
 
-        # Implicit int conversion
-        if t_type is int and f_type is not int:
+        # Implicit int/bv conversion
+        if (issubclass(t_type, (int, ht.BitVector)) and
+                not issubclass(f_type, (int, ht.BitVector))):
```
Comment (rsetaluri) [inline magma/bit.py:147]: nit: indent

File: magma/bits.py [hunk 3]
```diff
@@ -129,8 +129,10 @@ def __int__(self):
 
     @debug_wire
     def wire(self, other, debug_info):
-        if isinstance(other, IntegerTypes):
-            if other.bit_length() > len(self):
+        if isinstance(other, (IntegerTypes, BitVector)):
+            N = (other.bit_length() if isinstance(other, IntegerTypes)
+                 else len(other))
```
Comment (rsetaluri) [inline magma/bits.py:134]: nit
```suggestion
            N = (other.bit_length()
                 if isinstance(other, IntegerTypes)
                 else len(other))
```

File: magma/digital.py [hunk 4]
```diff
@@ -79,7 +80,7 @@ def __getitem__(cls, direction: Direction) -> 'DigitalMeta':
 
     def __call__(cls, value=None, *args, **kwargs):
         if value is not None:
-            if isinstance(value, (bool, IntegerTypes)):
+            if isinstance(value, (bool, IntegerTypes, )):
```
Comment (rsetaluri) [inline magma/digital.py:83]: nit: trailing space

File: magma/syntax/inline_combinational.py [hunk 5]
```diff
@@ -126,5 +126,6 @@ def exec(self, etree, stree, env):
                               result[len(self.wire_map):]):
             value_name = gen_free_name(ast.NameConstant(False), env)
             env.locals.update({value_name: value})
+            print(f"{astor.to_source(mutable(key)).rstrip()} = {value_name}")
```
Comment (rsetaluri) [inline magma/syntax/inline_combinational.py:129]: spurrious debug print?

File: magma/bit.py (hunk 2 above)
Comment (leonardt) [reply magma/bit.py:147]: What's the alternative here? If we line it up with the paren, it is 4 tabs in, so the block below will need to be indented, is that preferred?

Comment (rsetaluri) [reply magma/bit.py:147]: hm thats right, switching between looking at C++ and python gets tough 😩. Fine as is.

**Pull request merged**